jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/DoSHandler.java

```diff
@@ -255,372 +253,114 @@ protected void doStop() throws Exception
-                            _requestsInPreviousSample = requests;
-                            return true;
-                        }
-                        continue;
-                    }
-
-                    if (_state.compareAndSet(state, encodedSampleStart, requests + 1))
-                        return false;
-                }
-            }
-
-            @Override
-            public boolean isIdle(long now)
-            {
-                // We are idle if we roll over to a new sample period, with no requests
-                while (true)
-                {
-                    long state = _state.get();
-                    int sampleStartMs = AtomicBiInteger.getHi(state);
-                    int requests = AtomicBiInteger.getLo(state);
-
-                    long startSample = decodeNanoTime(sampleStartMs, now);
-                    if (NanoTime.elapsed(startSample, now) > _samplePeriod.toNanos())
-                    {
-                        if (_state.compareAndSet(state, encodeNanoTime(now), 0))
-                        {
-                            _requestsInPreviousSample = requests;
-                            return requests == 0;
-                        }
-                        continue;
-                    }
-                    return false;
-                }
-            }
-
-            @Override
-            public int getRequestsPerSecond(long now)
-            {
-                long state = _state.get();
-                int encodedSampleStart = AtomicBiInteger.getHi(state);
-                int requests = AtomicBiInteger.getLo(state);
-
-                long startSample = decodeNanoTime(encodedSampleStart, now);
-                long elapsed = NanoTime.elapsed(startSample, now);
-
-                // If we are more than 25% into a sample, then extrapolate from the current request count
-                if (elapsed > (_samplePeriod.toNanos() / 4))
-                    return Math.toIntExact(requests * TimeUnit.SECONDS.toNanos(1) / elapsed);
-
-                // Otherwise use the previous request count
-                return Math.toIntExact(_requestsInPreviousSample * TimeUnit.SECONDS.toNanos(1) / _samplePeriod.toNanos());
-            }
-
-            @Override
-            public String toString()
-            {
-                return "%s{%s}".formatted(super.toString(), _state.toString());
-            }
+            Tracker newTracker(String id);
         }
     }
 
     /**
-     * The Tracker implements the classic version of the <a link="https://en.wikipedia.org/wiki/Leaky_bucket">Leaky Bucket Algorithm</a>.
+     * The Tracker implements the classic <a href="https://en.wikipedia.org/wiki/Leaky_bucket">Leaky Bucket Algorithm</a>.
      */
-    public static class LeakingBucketTrackerFactory extends Tracker.Factory
+    public static class LeakingBucketTrackerFactory implements Tracker.Factory
     {
+        private final int _maxRequestsPerSecond;
+        private final int _bucketSize;
```

Comment:
This is a "max", so I'd prefer `_maxBucketSize`

Reply:
That doesn't make sense to me as a buckets size is independent of how full it is.   max bucket size  implies that we can have lots of buckets of different sizes or that buckets can change their size.

Reply:
If you want `max` in the name, then I think we should just drop the "bucket" analogy.  Using an analogy is a short hand, as we all know what a bucket is and that it has a fixed size and can be empty, full or somewhere in between.   If we start trying to over explain that, then we start doubting if the analogy is meaningful and then we may as well not have it.   We should call this `_threshold` or similar